controllers/topology/util.go

```diff
@@ -114,18 +114,30 @@ func (l *topologyReconcileLogger) Into(ctx context.Context) (context.Context, lo
 	return ctrl.LoggerInto(ctx, l.Logger), l
 }
 
-// KRef return a reference to a Kubernetes object in the same format used by kubectl commands (kind/name).
-type KRef struct {
+// KObj return a reference to a Kubernetes object in the same format used by kubectl commands (kind/name).
+type KObj struct {
```

Comment:
Query - why use this struct over the klog.KObj / Kobjs functions from klog? Kubernetes seems to prefer to use those functions directly. It would be good to document here why this was chosen as an alternative.

Reply:
Not 100% sure as it was introduced in @fabriziopandini 's error PR. Afaik because the klog struct uses the `namespace/name` format while we need `kind/name`

Reply:
You're right - I missed it there. It would be good to get the reasoning documented in a comment as it looks strange for anyone used to the k/k structured klogs implementation. This could also be changed to a function to make it easier to read and write (like the klog implementation), but I don't think it's a blocker.

Reply:
klog.KRef reports name and namespace; here we are reporting Kind and name (and assuming the namespace is always the same); also we need to stardardize logs/error around OwnerReferences

Reply:
Hm, what about the point about making this a func. I have no idea why it's a struct in klog and what we loose when I replace the struct by a simple func.

Reply:
Note: would be good if can get consensus and merge this PR relatively soon, because we have multiple big PRs inflight and it would be great if we could align to this already.

Reply:
I think it's fine to leave this as a struct for now and we can worry about making it a function later on.

Reply:
@killianmuldoon I added a comment to KObj. Afaik there is no comparable struct in klog to `KRef`.

Reply:
Great - I think that's good enough for context around this if we look to expand this across the code base in future.

Reply:
Yup, agree